calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -176,6 +181,708 @@ def test_validate_arguments(self, case, sys_exit_called):
+    @patch('calico_ctl.container.get_container_info_or_exit', autospec=True)
+    @patch('calico_ctl.container.client', autospec=True)
+    @patch('calico_ctl.container.netns', autospec=True)
+    def test_container_ip_remove_container_not_on_calico(
+            self, m_netns, m_client, m_get_container_info_or_exit,
+            m_get_pool_or_exit, m_enforce_root):
+        """
+        Test case when container is not networked into Calico
+        """
+        # Set up mock objects
+        m_get_container_info_or_exit.return_value = {
+            'Id': 666,
+            'State': {'Running': 1, 'Pid': 'Pid_info'}
+        }
+        m_client.get_endpoint.side_effect = KeyError
+
+        # Set up arguments to pass to method under test
+        container_name = 'container1'
+        ip = '1::1'
+        interface = 'interface'
+
+        # Call method under test
+        self.assertRaises(SystemExit, container.container_ip_remove,
+                          container_name, ip, interface)
+
+    @patch('calico_ctl.container.enforce_root', autospec=True)
+    @patch('calico_ctl.container.get_pool_or_exit', autospec=True)
+    @patch('calico_ctl.container.get_container_info_or_exit', autospec=True)
+    @patch('calico_ctl.container.client', autospec=True)
+    @patch('calico_ctl.container.netns', autospec=True)
+    def test_container_ip_remove_fail_updating_datastore(
+            self, m_netns, m_client, m_get_container_info_or_exit,
+            m_get_pool_or_exit, m_enforce_root):
+        """
+        Test case when client fails on updating endpoint in datastore
+        """
+        # Set up mock objects
+        m_get_container_info_or_exit.return_value = {
+            'Id': 666,
+            'State': {'Running': 1, 'Pid': 'Pid_info'}
+        }
+        ipv6_nets = set()
+        ipv6_nets.add(IPNetwork(IPAddress('1::1')))
+        m_endpoint = Mock(spec=Endpoint)
+        m_endpoint.ipv6_nets = ipv6_nets
+        m_client.get_endpoint.return_value = m_endpoint
+        m_client.update_endpoint.side_effect = KeyError
+
+        # Set up arguments to pass to method under test
+        container_name = 'container1'
+        ip = '1::1'
+        interface = 'interface'
+
+        # Call method under test
+        self.assertRaises(SystemExit, container.container_ip_remove,
+                          container_name, ip, interface)
+
+    @patch('calico_ctl.container.enforce_root', autospec=True)
+    @patch('calico_ctl.container.get_pool_or_exit', autospec=True)
+    @patch('calico_ctl.container.get_container_info_or_exit', autospec=True)
+    @patch('calico_ctl.container.client', autospec=True)
+    @patch('calico_ctl.container.netns', autospec=True)
+    def test_container_ip_remove_netns_error(
+            self, m_netns, m_client, m_get_container_info_or_exit,
+            m_get_pool_or_exit, m_enforce_root):
+        """
+        Test case when client fails on removing ip from interface through netns
```

Comment:
the client isn't failing here, just netns being unable to remove the ip from the interface.